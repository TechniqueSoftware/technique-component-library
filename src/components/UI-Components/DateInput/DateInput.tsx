import * as React from 'react';
import { TextField, TextFieldProps } from '../TextField/TextField';

export type DateInputProps = TextFieldProps;

export function DateInput(props: DateInputProps) {
  const propsToPassOn: TextFieldProps = {
    ...props,
    InputLabelProps: {
      ...props.InputLabelProps,
      shrink: true
    },
    type: 'date'
  };

  return <TextField {...propsToPassOn} />;
}
