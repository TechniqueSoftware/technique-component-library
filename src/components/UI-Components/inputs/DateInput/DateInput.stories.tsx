//@ts-nocheck
import * as React from 'react';
import ThemeProvider from '@material-ui/styles/ThemeProvider';
import * as moment from 'moment';
import { boolean, select, text } from '@storybook/addon-knobs';
import { createCUICStoriesOf } from '../../utils/utils';
import { MODULE_NAME } from '../../constants';
import { DateInputProps, DateInput } from '../../DateInput/DateInput';
import clubOsTheme from '../../storyThemes';

type DateInputStoryProps = {
  label: string;
  variant: 'standard' | 'outlined' | 'filled' | undefined | any;
  disabled: boolean;
};

function DateInputStory(props: DateInputStoryProps) {
  const today = moment().format('YYYY-MM-DD');
  const { label, variant, disabled } = props;
  const [value, setValue] = React.useState(today);

  const dateInputProps: DateInputProps = {
    label,
    value,
    disabled,
    variant,
    id: 'Date Text Field',
    onChange: (event) => setValue(event.target.value),
    min: moment('0001-01-01').format('YYYY-MM-DD'),
    max: moment('9999-12-31').format('YYYY-MM-DD')
  };

  return (
    <div>
      <ThemeProvider theme={clubOsTheme}>
        <DateInput {...dateInputProps}/>
      </ThemeProvider>
    </div>
  );
}

createCUICStoriesOf(MODULE_NAME.INPUTS, module)
  .addCUICStory({
    render: () => {
      const label = text('Label', 'Date Text Field');
      const variant = select('variant', ['standard', 'filled', 'outlined'], 'outlined');
      const disabled = boolean('disabled', false);
      return (<DateInputStory
        label={label}
        variant={variant}
        disabled={disabled}
      />);
    },
    name: 'DateInput',
    notes: require('./notes.md').default
  });
