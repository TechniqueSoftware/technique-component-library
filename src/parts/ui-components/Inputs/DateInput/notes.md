# DateInput
This component wraps Material-UI's [TextField](https://material-ui.com/api/form-control/) and passes type='date' to
create a textField component used strictly for date values. Accepts and passes values in the 'YYYY-MM-DD' format.

## JIRA
Related jira ticket: https://clubos.atlassian.net/browse/CR-747
