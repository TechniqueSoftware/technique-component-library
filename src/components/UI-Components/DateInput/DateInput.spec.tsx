import * as React from 'react';
import { DateInput } from './DateInput';
import { shallow } from 'enzyme';

describe('Date Input', () => {
  test('should render with minimal props', () => {
    const shallowWrapper = shallow(<DateInput />);
    expect(shallowWrapper.debug()).toMatchSnapshot();
  });
});
