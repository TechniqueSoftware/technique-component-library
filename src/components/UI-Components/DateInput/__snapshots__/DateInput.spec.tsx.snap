// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date Input should render with minimal props 1`] = `"<TextField InputLabelProps={{...}} type=\\"date\\" />"`;
